### DISCUSSION TITLE
matplotlib.cpp library in Visual Studio -> missing coloring of plain text in plots

### DISCUSSION THREAD

**Discussion by udofritz** [community/community#135125] [Programming Help]
### Body

I want to plot colored text by the Python-like statement
void text(Numeric x, Numeric y, const std::string &s = ""),
but the keyword parameter is missing in matplotlib.cpp as provided for the plot-statement ! 

Who knows about any solution or improvement of this Python-wrapper for C++ ?

UFM

### Guidelines

- [X] I have read and understood this category's [guidelines](https://github.com/orgs/community/discussions/51384) before making this post.

**Comment by geniuszly** [comment]
In Python's Matplotlib, you can easily add text with color by using the `color` keyword argument, like so:
```python
plt.text(x, y, "text", color="blue")
```
However, in the C++ wrapper matplotlib-cpp, this functionality may not be directly implemented or may not support keyword arguments in the same way. But, you can modify the `matplotlib-cpp` library to support additional keyword arguments like `color` for the `text()` function. Here’s a basic example of how you might extend the `text()`method:
```cpp
void text(double x, double y, const std::string &s = "", const std::string &color = "black") {
    PyObject* kwargs = PyDict_New();
    PyDict_SetItemString(kwargs, "color", PyString_FromString(color.c_str()));
    PyObject* args = PyTuple_Pack(3, PyFloat_FromDouble(x), PyFloat_FromDouble(y), PyString_FromString(s.c_str()));
    PyObject* res = PyObject_Call(detail::_interpreter::get().s_python_function_text, args, kwargs);
    if (!res) throw std::runtime_error("Call to text() failed");
    Py_DECREF(args);
    Py_DECREF(kwargs);
    Py_XDECREF(res);
}
```
This function extends the existing ` text ` method to include a color parameter. It constructs a Python dictionary (`kwargs`) that holds the keyword arguments, allowing you to set the color.

**Comment by udofritz** [comment]
Problem resolved!
[TEXT_COLOR.pdf](https://github.com/user-attachments/files/16599386/TEXT_COLOR.pdf)
